x/qgb/abci.go

```diff
@@ -9,20 +9,37 @@ import (
 
 // EndBlocker is called at the end of every block
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
-	handleValsetRequest(ctx, k)
 	handleDataCommitmentRequest(ctx, k)
+	handleValsetRequest(ctx, k)
 }
 
 func handleDataCommitmentRequest(ctx sdk.Context, k keeper.Keeper) {
-	if ctx.BlockHeight()%int64(types.DataCommitmentWindow) != 0 {
-		return
+	if ctx.BlockHeight() != 0 && ctx.BlockHeight()%int64(types.DataCommitmentWindow) == 0 {
+		dataCommitment, err := k.GetCurrentDataCommitment(ctx)
+		if err != nil {
+			panic(sdkerrors.Wrap(err, "coudln't get current data commitment"))
+		}
+		k.StoreAttestation(ctx, &dataCommitment)
+		k.SetLatestAttestationNonce(ctx, dataCommitment.Nonce)
 	}
-	k.SetDataCommitmentRequest(ctx)
 }
 
 func handleValsetRequest(ctx sdk.Context, k keeper.Keeper) {
-	//get the last valsets to compare against
-	latestValset := k.GetLatestValset(ctx)
+	// FIXME can this be done here?
+	// We need to initialize this value in the begining...
+	if !k.CheckLatestAttestationNonce(ctx) {
+		k.SetLatestAttestationNonce(ctx, 0)
+	}
```

Comment:
let's create an issue for initializing these values in genesis

Reply:
https://github.com/celestiaorg/celestia-app/issues/502